package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua

```diff
@@ -212,15 +212,25 @@ end
 local function get_wlan_mac_from_driver(uci, radio, vif)
 	local primary = sysconfig.primary_mac:lower()
 
-	local i = 1
-	for addr in get_addresses(uci, radio) do
-		if addr:lower() ~= primary then
-			if i == vif then
-				return addr
-			end
+	local addresses = {}
+	for address in get_addresses(uci, radio) do
+		if address:lower() ~= primary then
+			table.insert(addresses, address)
+		end
+	end
 
-			i = i + 1
+	-- Make sure we have at least 4 addresses
+	if table.getn(addresses) < 4 then
+		return nil
+	end
+
+	local i = 1
+	for _, addr in ipairs(addresses) do
```

Comment:
We can just write `for i, addr in ...` and avoid counting ourselves.

Reply:
done